examples/dependency-parsing/run_udp.py

```diff
@@ -0,0 +1,342 @@
+    )
+    logger.warning(
+        "Process rank: %s, device: %s, n_gpu: %s, distributed training: %s, 16-bits training: %s",
+        training_args.local_rank,
+        training_args.device,
+        training_args.n_gpu,
+        bool(training_args.local_rank != -1),
+        training_args.fp16,
+    )
+    logger.info("Training/evaluation parameters %s", training_args)
+
+    # Set seed
+    set_seed(training_args.seed)
+
+    # Prepare for UD dependency parsing task
+    labels = UD_HEAD_LABELS
+    label_map: Dict[int, str] = {i: label for i, label in enumerate(labels)}
+    num_labels = len(labels)
+
+    config = AutoConfig.from_pretrained(
+        model_args.config_name if model_args.config_name else model_args.model_name_or_path,
+        num_labels=num_labels,
+        id2label=label_map,
+        label2id={label: i for i, label in enumerate(labels)},
+        cache_dir=model_args.cache_dir,
+        pad_token_id=-1,
+    )
+
+    if model_args.is_japanese:
+        assert model_args.mecab_dir is not None
+        assert model_args.mecab_dic_dir is not None
+
+    tokenizer = AutoTokenizer.from_pretrained(
+        model_args.tokenizer_name if model_args.tokenizer_name else model_args.model_name_or_path,
+        cache_dir=model_args.cache_dir,
+        use_fast=model_args.use_fast,
+        do_lower_case=model_args.do_lower_case,
+        add_prefix_space=True,  # Used e.g. for RoBERTa
+        mecab_kwargs={"mecab_option": f"-r {model_args.mecab_dir} -d {model_args.mecab_dic_dir}"}
+        if model_args.is_japanese
+        else None,
+    )
+
+    # The task name (with prefix)
+    task_name = "ud_" + data_args.task_name
+    language = adapter_args.language
+
+    model = AutoModelWithHeads.from_pretrained(
+        model_args.model_name_or_path,
+        config=config,
+        cache_dir=model_args.cache_dir,
+    )
+    model.add_dependency_parsing_head(
+        task_name,
+        num_labels=num_labels,
+        id2label=label_map,
+    )
+
+    if model_args.leave_out_twelvth:
+        logger.info("Leaving out 12")
+        leave_out = [11]
+    else:
+        leave_out = []
+
+    # Setup adapters
+    if adapter_args.train_adapter:
+        # check if adapter already exists, otherwise add it
+        if task_name not in model.config.adapters:
+            # resolve the adapter config
+            adapter_config = AdapterConfig.load(
+                adapter_args.adapter_config,
+                non_linearity=adapter_args.adapter_non_linearity,
+                reduction_factor=adapter_args.adapter_reduction_factor,
+                leave_out=leave_out,
+            )
+            # load a pre-trained from Hub if specified
+            if adapter_args.load_adapter:
+                model.load_adapter(
+                    adapter_args.load_adapter,
+                    config=adapter_config,
+                    load_as=task_name,
+                    leave_out=leave_out,
```

Comment:
Why is the leave out specified in the config and during loading?

Reply:
There's a slight semantic difference between the two: Setting it in the config searches for adapters that come with the layer dropped, setting it during loading drops it dynamically. I think the first specification is probably not needed.